linera-rpc/src/node_provider.rs

```diff
@@ -32,14 +39,22 @@ impl ValidatorNodeProvider for NodeProvider {
 
     fn make_node(&self, address: &str) -> anyhow::Result<Self::Node, NodeError> {
         let address = address.to_lowercase();
+        let mut pool = self.client_pool.lock().unwrap();
```

Comment:
I don't think this can panic and poison the lock, that's why the unwrap is here.